src/libs/actions/PersonalDetails.js

```diff
@@ -186,6 +199,36 @@ function getFromReportParticipants(reports) {
         });
 }
 
+
+/**
+ * Sets the personal details object for the current user
+ *
+ * @param {Object} details
+ */
+function setPersonalDetails(details) {
+    API.PersonalDetails_Update({details: JSON.stringify(details)});
```

Comment:
NAB, I wasn't going to say anything the first time, but seeing the `JSON.stringify()` again makes me wonder if this is the best place to do this. Anytime we have an API parameter that is an `{Object}` we will need to remember to call `JSON.stringify()`. It would be nice (but maybe not totally necessary yet) if `HttpUtils.xhr()` was smart enough to stringify any object params.

https://github.com/Expensify/Expensify.cash/blob/64349672d6fde67d2251de791d7694c0be91b784/src/libs/HttpUtils.js#L47-L51

maybe by doing something like this...

```js
formData.append(key, _.isObject(val) ? JSON.stringify(val) : val)
```

Reply:
This is a good point, I think we should add it so we don't have to `JSON.stringify` all over the place.